### ISSUE TITLE
Implement denominator computation in python

### ISSUE THREAD

**Issue by vimalmanohar** [vimalmanohar/pychain-old#1]
Create a new class (maybe derived from `Function`) with the arguments:

#### [in] chain_training_options
A struct containing various options relevant to LF-MMI training:
* leaky_hmm_coefficient [0.1]

#### [in] transition_matrix
SparseTensor of size N x N x D, where
N = num states in denominator graph
D = num pdfs
T = chunk size
B = minibatch size
A sparse tensor representation of the transition probabilities of the arcs in the denominator graph. Depending on the implementation, we might require this to be stored as real numbers or in log format.

#### [in] initial_probs
Tensor of size N x 1
A tensor containing the initial probabilities for the states in the denominator graph. This is also needed for leaky-hmm computation.

#### [in] nnet_outputs 
Tensor of size D x T x B
A tensor of neural network activations.

#### [out] nnet_gradients
Tensor of size D x T x B
Gradients computed by the network. In a python implementation, this will be computed using autograd backward() function.
